### PR TITLE
Stepper Flow: Build the intent capture step

### LINKED ISSUE #60810: Use the Intent Screen upcoming package to build the intent capture step in Stepper
### Details

Once https://github.com/Automattic/wp-calypso/issues/60809 is done. It should be easy to build the intent capture step in Stepper.

- Before building the step inside Stepper, explore a bit whether it makes sense to make the step itself a package that lives next to `intent-screen`. Steps are highly customized, so it might not be a good option, but it's worth it to explore making the step also reusable.
- Steps in Stepper are just react-router routes. So to add a step, please just a route under add nice path name. (e.g.: `intent`). Check this for [inspo](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/gutenboarding/onboarding-block/edit.tsx#L214-L224).

### Checklist

_No response_

### Related

Blocked by #60809

### PR DESCRIPTION
**Pull request by escapemanuele**
#### Changes proposed in this Pull Request

Build the intent capture step

#### Testing instructions

* Checkout branch
* `yarn start`
* Go to `http://calypso.localhost:3000/stepper/intent`

Related to #60810
Fixes #60810

### CODE REVIEW

File: client/landing/stepper/declarative-flow/example-flow.ts [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 			}
 		};
 		const goNext = goBack;
-		return { goNext, goBack };
+		const goToPage = ( page: StepPath ) => {
+			navigate( page );
+		};
+		return { goNext, goBack, goToPage };
 	},
 };
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/example-flow.ts]: This file should remain untouched as an example.

Can you create another flow called `siteSetup` please?

File: client/landing/stepper/declarative-flow/internals/types.ts [hunk 2]
```diff
@@ -12,6 +12,10 @@ export type NavigationControls = {
 	 * Call this function if you want to go to the proceed down the flow.
 	 */
 	goNext: () => void;
+	/**
+	 * Call this function if you want to jump to a certain step.
+	 */
+	goToPage: ( page: StepPath ) => void;
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/types.ts]:
```suggestion
	goToStep: ( step: StepPath ) => void;
```

**Review by alshakero — CHANGES_REQUESTED** [review]
That's pretty good! Left some comments. I think we can start with the CSS now. It would be great if you could make it identical to https://wordpress.com/start/setup-site/intent?siteSlug=SITE_URL&loading_ellipsis=1

File: client/landing/stepper/declarative-flow/internals/steps-repository/build.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * The build step
+ */
+const BuildStep: Step = function BuildStep( { navigation } ) {
+	const { __ } = useI18n();
+	const { goNext, goBack } = navigation;
+
+	return (
+		<div>
+			<h1>Domains step</h1>
```
Comment (arcangelini) [inline client/landing/stepper/declarative-flow/internals/steps-repository/build.tsx]: I know this is just a placeholder, but I think this should be: `<h1>Build step</h1>`

**Review by arcangelini — COMMENTED** [review]
Everything is working for me. I had two thoughts that I don't think apply to the actual step but more to the component.

1. In the name change PR the `intent-screen.tsx` became `onboarding-component.tsx` is this correct? Or should it actually be `intent-screen.tsx` because that is the component its exporting.
2. Will there always be a need for `SelectAltItem` or would it be beneficial to make this optional?

**Review by alshakero — APPROVED** [review]
Looks perfect! Addressed some issues directly to save time.

**Pull request merged**